### PR TITLE
Add Application Toolkit Package

### LINKED ISSUE #2848: Introduce an Application Toolkit Package
# Proposal: Introduce an Application Toolkit Package

# Summary
We are introducing an Application Toolkit Meta-Package to simplify the installation of all the packages that are important for a developer to create and iterate on Tanzu workloads. 

# Problem Statement

The introduction of carvel packaging technology has significantly improved the installation experience of tools on Kubernetes. We've also seen a wide adoption of the packaging technology for several tools on TCE. 

Installing each of the individual packages manually can be cumbersome and error-prone. 

# Proposal

**Meta-Package**

A meta-package is a set of components (Carvel packages). Once that meta-package is defined, a profile can be implemented based on the need to enable an intent-based installation experience of some or all of the specified packages.

Further details regarding the Metapackage as a concept has been introduced in the [Design Doc](https://github.com/vmware-tanzu/community-edition/pull/2911)

We are introducing an **App-Platform** (name can still change) metapackage which will initially consists of the below packages. This will expand in the future. 

**Phase 1 Packages**

| Packages                |
| ----------------------- |
| Contour                 |
| Cert-manager            |
| Source Controller       |
| knative-serving         |
| kpack                   |
| Cartographer            |

**User Experience**

Here is the flow of the installation experience using the developer profile of Application Toolkit Meta-Package 

1. Listing all the packages as they relate to Application Toolkit (there can be additional non-Application Toolkit packages here)  

`tanzu package available list`

```
NAME                                     DISPLAY-NAME             SHORT-DESCRIPTION
  
app-toolkit.tanzu.vmware.com             Application Toolkit             Package to install a set of Application Toolkit components
contour.tanzu.vmware.com                 Contour                  Used as a Ingress Controller
cert-manager.tanzu.vmware.com            Cert-Manager             Used to install certificates
source-controller.tanzu.vmware.com       Source Controller        Used to watch the git repositories
knative-serving.tanzu.vmware.com         Knative Serving          Used to convert images to Knative services
kpack.tanzu.vmware.com                   Kpack                    Used to convert source code into images
cert-injection.webhook.tanzu.vmware.com  Cert-Injection Webhook   Used to inect certificates into the running containers
cartographer.tanzu.vmware.com            Cartographer             Used to create software supply-chains
ootb-basic-supply-chain.tanzu.vmware.com OOTB Basic Supply Chain  Used to install a basic supply chain going from git repository to running Knative Service. 
```

2. Installing a profile
* List version information for the package by running:

```
tanzu package available list app-toolkit.tanzu.vmware.com 
```
* Create a `app-toolkit-config-values.yaml` file. This configuration file will have configuration required to deploy foo and the subordinate packages. 

**Example Config for Application Toolkit**
```
kpack: 
  kp_default_repository: "KP-DEFAULT-REPO"
  kp_default_repository_username: "KP-DEFAULT-REPO-USERNAME"
  kp_default_repository_password: "KP-DEFAULT-REPO-PASSWORD"
supply_chain: basic

ootb_supply_chain_basic:
  registry:
    server: "SERVER-NAME"
    repository: "REPO-NAME"
  gitops:
    ssh_secret: ""
```
**Note** *Accurate configuration variables will be defined later*

* Install the metapackage
```
tanzu package install app-toolkit -p app-toolkit.tanzu.vmware.com -v 1.0.0 --values-file app-toolkit-config-values.yml
```

See Also
- 
- [Proposal: Introduce Tanzu Workload Creation Capability](https://github.com/vmware-tanzu/community-edition/issues/2847)

### PR DESCRIPTION
**Pull request by mateogo42**
## What this PR does / why we need it

This PR introduces the Application Toolkit metapackage (app-toolkit) which enables developers to deploy a set of TCE components used for builiding and deploying workloads. When installing app-toolkit the following packages get installed:
- Cartographer
- Cert-manager
- Contour
- FluxCD Source Controller
- Knative Serving
- Kpack

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Add app-toolkit v0.1.0 package
```

## Which issue(s) this PR fixes

Closes: #2848

### CODE REVIEW

**Comment by joshrosso** [comment]
@mateogo42 can you please squash these commits into 1 detailed commit about adding this feature?

File: addons/packages/app-toolkit/0.1.0/bundle/config/contour.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  namespace: tanzu-package-repo-global
+  annotations:
+    kapp.k14s.io/change-group: contour
+    kapp.k14s.io/change-rule.serviceaccount: "delete before deleting serviceaccount"
+spec:
+  serviceAccountName: app-toolkit-install-sa
+  packageRef:
+    refName: contour.community.tanzu.vmware.com
+    versionSelection:
+      constraints: 1.19.1
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/bundle/config/contour.yaml:21]: I would recommend the latest version of Contour, [1.20.1](https://github.com/projectcontour/contour/releases/tag/v1.20.1), as it contains a CVE fix

**Review by seemiller — CHANGES_REQUESTED** [review]

File: docs/developer/app-toolkit.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+The OSS App Package concept is comparable to proprietary Tanzu Application Platform, although reduced in scope.
+
+### Configuration of App Toolkit and Included TCE Packages.
+
+Configuration is deferred to the included TCE package. For instance, if end users are interested in changing the default behavior of, say, knative-serving, they will learn about and provide configuration for knative-serving, not app-toolkit.
+
+### Similarities and Differences to TCE Packaging of Upstream Projects
+
+TCE provides guidance about how to include other upstream projects into TCE - e.g. kpack, knative, etc., as described in the [TCE Documentation](https://tanzucommunityedition.io/docs/latest/package-creation-step-by-step/). Principly it describes how to
```
Comment (seemiller) [inline docs/developer/app-toolkit.md:29]:
```suggestion
TCE provides guidance about how to include other upstream projects into TCE - e.g. kpack, knative, etc., as described in the [TCE Documentation](https://tanzucommunityedition.io/docs/latest/package-creation-step-by-step/). Principally it describes how to
```

**Review by seemiller — CHANGES_REQUESTED** [review]

File: docs/developer/app-toolkit.md (hunk 2 above)
Comment (seemiller) [reply docs/developer/app-toolkit.md:29]: Also need to fix up all the lint warnings from the checks - those are too numerous to provide suggestions for.

File: addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+func main() {
+	fmt.Println("STARTING TEST")
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go:15]: Could a `README.md` be created in the `test` directory that describes how to run/execute the tests?

**Review by seemiller — CHANGES_REQUESTED** [review]

File: docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
+| contour | contour.community.tanzu.vmware.com | 1.19.1 |
+| fluxcd-source-controller | fluxcd-source-controller.community.tanzu.vmware.com | 0.21.2 |
+| knative_serving | knative-serving.community.tanzu.vmware.com | 1.0.0 |
+| kpack | kpack.community.tanzu.vmware.com | 4.5.1 |
```
Comment (seemiller) [inline docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md:24]:
```suggestion
| kpack | kpack.community.tanzu.vmware.com | 0.5.1 |
```

**Review by seemiller — CHANGES_REQUESTED** [review]

**Comment by kartiklunkad26** [comment]
> Are we still using the term meta-package? These 2 sections read very similar, should they be combined, or the meta-package section deleted?

> If app toolkit is the preferred term, we should clean up all references to meta-package.

Meta-package is the concept. App Toolkit is the first of those meta-packages in TCE. IMO, explaining the concept of meta-package is helpful and distinct from the actual App Toolkit meta-package.

File: docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+1. Create a cluster named demo-local exposing 80 and 443 ports.
+
+    ```
+    tanzu unmanaged-cluster create demo-local -p 80:80 -p 443:443
+    ```
+
+1. Follow the steps listed above in the "Local docker installation without load balancer" section
+
+TODO: Update the file links to Github links
```
Comment (seemiller) [inline docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md:103]: Address the TODO - Update the file links to Github links

**Review by seemiller — CHANGES_REQUESTED** [review]

File: docs/developer/app-toolkit.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+## Concepts
+
+### Meta-package
+
+A meta-package is defined as a TCE package that installs a set of other already present TCE packages. It is not intended to add new functionality on its own, but deploy a set of existing functionalities to achieve a particular outcome.
+
+### App Toolkit 
+
+App Toolkit is an instance of a meta-package with a goal of allowing TCE users to deploy a set of TCE packages sufficient to deploy a workload. App Toolkit itself provides little/no novel functionality, but comprises an opinion about what TCE packages are required to achieve the outcome.
```
[the remark this answers is not in the document]
Comment (krisapplegate) [reply docs/developer/app-toolkit.md:19]: I think this speaks to our distinction that the meta-package capability is distinct and standalone apart from the app-toolkit. There's the potential for other meta-packages for unrelated reasons, so making sure to communicate this distinction I think is key. It's also why there were two feature requests (meta-package and app-toolkit).

File: addons/packages/app-toolkit/0.1.0/bundle/config/contour.yaml (hunk 1 above)
Comment (krisapplegate) [reply addons/packages/app-toolkit/0.1.0/bundle/config/contour.yaml:21]: +1 - it shouldn't impact us and only benefits us.

File: docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+1. Set up kpack and required resources
+
+    ```
+    ytt --data-values-file test-supply-chain-values.yaml --ignore-unknown-comments -f https://raw.githubusercontent.com/cgsamp/tanzu-simple-web-app/main/config/example_sc/rbac.yaml | kubectl apply -f -
```
Comment (seemiller) [inline docs/site/content/docs/latest/package-readme-app-toolkit-0.1.0.md:108]: Not sure if this is related to the TODO comment, but there are no files in the package or examples of `test-supply-chain-values.yaml`. Running these `ytt` obviously fail.

**Review by seemiller — CHANGES_REQUESTED** [review]

**Comment by stmcginnis** [comment]
Please squash commits when ready.

File: addons/packages/app-toolkit/0.1.0/test/Makefile [hunk 8]
```diff
@@ -0,0 +1,2 @@
+get-deps: ## Get all dependencies
```
Comment (stmcginnis) [inline addons/packages/app-toolkit/0.1.0/test/Makefile:1]: There are a few other targets that are expected. You can probably copy something like https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/antrea/1.2.3/test/Makefile

File: addons/packages/app-toolkit/0.1.0/test/go.mod [hunk 9]
```diff
@@ -0,0 +1,5 @@
+module github.com/vmware-tanzu/community-edition/addons/packages/app-toolkit/0.1.0/test
+
+go 1.16
```
Comment (stmcginnis) [inline addons/packages/app-toolkit/0.1.0/test/go.mod:3]: This should be go 1.17.

File: docs/developer/app-toolkit.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (stmcginnis) [reply docs/developer/app-toolkit.md]: > the meta-package capability

This has been a point of some confusion, at least for me, so wondering if you could help clarify.

The capability to have a meta-package is something that is built into the carvel toolset, so is a capability that is available in TCE. But a few of the docs have talked about it as a new functionality being added to TCE. Can you clarify what this is and why it is something separate from introducing a package containing the app toolkit packages?

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+## Components
+
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
+| contour | contour.community.tanzu.vmware.com | 1.19.1 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:21]:
```suggestion
| contour | contour.community.tanzu.vmware.com | 1.20.1 |
```
Thanks for bumping Contour to 1.20.1, just need one more place...

**Review by seemiller — CHANGES_REQUESTED** [review]

**Comment by seemiller** [comment]
Thanks for all the updates!

Still need to address all of the lint/mdlint issues, and we need the e2e test to pass.

**Comment by kartiklunkad26** [comment]
Given the E2E tests finally worked. @seemiller what are the remaining changes? 

1. Linting errors
2. Squashing commits

Anything else?

**Comment by seemiller** [comment]
correct with lint and squash. I'm still giving it a final once over...

File: addons/packages/app-toolkit/0.1.0/test/Makefile [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+endif
+
+get-deps: ## Get all dependencies
+	go mod download
+
+test: ## Run unit testing suite
+	@echo "TODO: implement running unit tests"
+
+e2e-test: ## Run e2e testing suite
+	@echo "TODO: implement running e2e tests"
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/test/Makefile:25]:
```suggestion
	go run app-toolkit-test.go
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
+# (Experimental) Application Toolkit
+
+Application Toolkit package is a meta-package that helps user install a collection of packages for creating, iterating and managing applications
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:3]:
```suggestion
Application Toolkit package is a meta-package that installs a collection of packages for creating, iterating and managing applications
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+## Supported Providers
+
+The following table shows the providers this package can work with.
+
+| Docker |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:9]: Does this package only work on Docker? I've only tested locally with an unmanaged-cluster, should we try AWS or Azure?

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+kpack:
+  kp_default_repository: https://index.docker.io/v1/
+  kp_default_repository_username: [your username]
+  kp_default_repository_password: [your password]
+```
+
+## Usage Example
+
+This example illustrates creating a simple spring boot web app on a locally deployed TCE cluster using docker.
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:97]:
```suggestion
This example illustrates creating a simple Spring Boot web app on a locally deployed TCE cluster using Docker.
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+
+1. Begin with docker and the tanzu cli installed on your development machine.
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:99]:
```suggestion
1. Begin with Docker and the Tanzu CLI installed on your development machine.
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+
+  For best performance, configure Docker with at least 8GB of RAM.
+
+1. Create a cluster named demo-local exposing 80 and 443 ports.
+
+    ```shell
+    tanzu unmanaged-cluster create demo-local -p 80:80 -p 443:443
+    ```
+
+1. Follow the steps listed above in the "Local docker installation without load balancer" section
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:109]:
```suggestion
1. Follow the steps listed above in the "Local Docker installation without load balancer" section
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+For example:
+
+```yaml
+excluded_packages:
+  - contour.community.tanzu.vmware.com
+  - cert-manager.community.tanzu.vmware.com
+```
+
+### Local docker installation without load balancer
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:72]:
```suggestion
### Local Docker installation without load balancer
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+
+To deploy contour with a ClusterIP configuration, and knative to serve at localhost, add these settings to a `values.yaml` file. Also, add a docker registry for kpack to store produced container images.
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:74]:
```suggestion
To deploy Contour with a ClusterIP configuration, and Knative to serve at localhost, add these settings to a `values.yaml` file. Also, add a Docker registry for kpack to store produced container images.
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+|:---:|
+| ✅  |
+
+## Components
+
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:19]:
```suggestion
| Cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
+| contour | contour.community.tanzu.vmware.com | 1.20.1 |
+| fluxcd-source-controller | fluxcd-source-controller.community.tanzu.vmware.com | 0.21.2 |
+| knative_serving | knative-serving.community.tanzu.vmware.com | 1.0.0 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:23]:
```suggestion
| Knative Serving | knative-serving.community.tanzu.vmware.com | 1.0.0 |
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+## Components
+
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
+| contour | contour.community.tanzu.vmware.com | 1.20.1 |
+| fluxcd-source-controller | fluxcd-source-controller.community.tanzu.vmware.com | 0.21.2 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:22]:
```suggestion
| Flux CD Source Controller | fluxcd-source-controller.community.tanzu.vmware.com | 0.21.2 |
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 22]
```diff
@@ -0,0 +1,141 @@
...
+
+## Components
+
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
+| contour | contour.community.tanzu.vmware.com | 1.20.1 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:21]:
```suggestion
| Contour | contour.community.tanzu.vmware.com | 1.20.1 |
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+| ✅  |
+
+## Components
+
+* App Toolkit version: 0.1.0
+
+| Name | Package | Version |
+|---------|----------------|---|
+| cartographer | cartographer.community.tanzu.vmware.com | 0.2.2 |
+| cert_manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:20]:
```suggestion
| cert-manager | cert-manager.community.tanzu.vmware.com | 1.6.1 |
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+
+## Installation
+
+### Prerequisites
+
+* The following steps require the Tanzu CLI app plugin. See [here](https://github.com/vmware-tanzu/apps-cli-plugin#getting-started) on how to install.
+
+### Steps
+
+The App Toolkit package is a meta-packages - a collection of packages - to build and deploy workloads. It requires that TCE is installed and the tanzu CLI is available.
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:44]:
```suggestion
The App Toolkit package is a meta-packages - a collection of packages - to build and deploy workloads. It requires that TCE is installed and the Tanzu CLI is available.
```

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+    ytt --data-values-file values.yaml --ignore-unknown-comments -f https://raw.githubusercontent.com/cgsamp/tanzu-simple-web-app/main/config/example_sc/builder.yaml | kubectl apply -f -
+    ```
+
+1. Deploy the example supply chain
+
+    ```shell
+    ytt --data-values-file values.yaml --ignore-unknown-comments -f https://raw.githubusercontent.com/cgsamp/tanzu-simple-web-app/main/config/example_sc/supplychain.yaml | kubectl apply -f -
+    ```
+
+1. Deploy the tanzu workload
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:125]:
```suggestion
1. Deploy the Tanzu workload
```

**Review by seemiller — CHANGES_REQUESTED** [review]
Lots of capitalization changes for consistency

**Comment by kartiklunkad26** [comment]
> Does this package only work on Docker? I've only tested locally with an unmanaged-cluster, should we try AWS or Azure?

The App Toolkit package should work with AWS or Azure, but we haven’t tested it. Hence marking as experimental package.

File: addons/packages/app-toolkit/0.1.0/README.md [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+| Contour | contour.community.tanzu.vmware.com | 1.20.1 |
+| Flux CD Source Controller | fluxcd-source-controller.community.tanzu.vmware.com | 0.21.2 |
+| Knative-serving | knative-serving.community.tanzu.vmware.com | 1.0.0 |
+| kpack | kpack.community.tanzu.vmware.com | 0.5.1 |
+
+## Configuration
+
+| Config | Values | Description |
+|--------|--------|-------------|
+| cert_manager | | [See cert_manager documentation](https://tanzucommunityedition.io/docs/latest/package-readme-cert-manager-1.6.1/#configuration)|
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/README.md:30]:
```suggestion
| cert_manager | | [See cert-manager documentation](https://tanzucommunityedition.io/docs/latest/package-readme-cert-manager-1.6.1/#configuration)|
```

**Review by seemiller — CHANGES_REQUESTED** [review]

File: addons/packages/app-toolkit/0.1.0/test/wait_for_app_toolkit.sh [hunk 27]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/test/wait_for_app_toolkit.sh:2]: # Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

**Review by seemiller — CHANGES_REQUESTED** [review]

File: addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go [hunk 28]
```diff
@@ -0,0 +1,182 @@
...
+	fmt.Println("\nTEST STEP Install TCE and wait for ready")
+	runCommand("tanzu uc create app-toolkit-test -p 80:80 -p 443:443")
+	pollCommand("kubectl get nodes", " Ready", 10)
+	fmt.Println("TEST STEP Install TCE and wait for ready OK")
+
+	fmt.Println("\nTEST STEP Install dev package PRERELEASE ONLY")
+	runCommand("tanzu package repository update projects.registry.vmware.com-tce-main-v0.11.0 -n tanzu-package-repo-global --url projects.registry.vmware.com/tce/main:v0.11.0-alpha.3568.3")
+	fmt.Println("TEST STEP Install dev package PRERELEASE ONLY OK")
+
+	fmt.Println("\nTEST STEP Install app-toolkit")
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go:47]:
```suggestion
	fmt.Println("TEST STEP Install TCE and wait for ready OK")

	fmt.Println("\nTEST STEP Install app-toolkit")
```
Let's back out updating this package repo

**Review by seemiller — CHANGES_REQUESTED** [review]
Let's fix this URL lint warning since it is in a doc being added by this PR. The other failing check is in a doc not touched by this PR, so we'll fix that in another issue.

File: addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go [hunk 29]
```diff
@@ -0,0 +1,182 @@
...
+	fmt.Println("\nTEST STEP Install TCE and wait for ready")
+	runCommand("tanzu uc create app-toolkit-test -p 80:80 -p 443:443")
+	pollCommand("kubectl get nodes", " Ready", 10)
+	fmt.Println("TEST STEP Install TCE and wait for ready OK")
+
+	fmt.Println("\nTEST STEP Install dev package PRERELEASE ONLY")
+	runCommand("tanzu package repository update tkg-core-repository -n tanzu-package-repo-global --url projects.registry.vmware.com/tce/main:v0.11.0-alpha.3568.3")
+	fmt.Println("TEST STEP Install dev package PRERELEASE ONLY OK")
+
+	fmt.Println("\nTEST STEP Install app-toolkit")
```
Comment (seemiller) [inline addons/packages/app-toolkit/0.1.0/test/app-toolkit-test.go:47]:
```suggestion
	fmt.Println("TEST STEP Install TCE and wait for ready OK")

	fmt.Println("\nTEST STEP Install app-toolkit")
```

**Review by seemiller — CHANGES_REQUESTED** [review]

**Review by seemiller — APPROVED** [review]

**Pull request merged**